core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplanation.java

```diff
@@ -141,6 +150,11 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             // If we have unassigned info, show that
             if (unassignedInfo != null) {
                 unassignedInfo.toXContent(builder, params);
+                long delay = unassignedInfo.getLastComputedLeftDelayNanos();
+                builder.field("allocation_delay", TimeValue.timeValueNanos(delay));
+                builder.field("allocation_delay_ms", TimeValue.timeValueNanos(delay).millis());
+                builder.field("remaining_delay", TimeValue.timeValueNanos(remainingDelayNanos));
+                builder.field("remaining_delay_ms", TimeValue.timeValueNanos(remainingDelayNanos).millis());
```

Comment:
I believe this is only going to get updated when the master node changes (there is a lot of weird things with this regarding nanos and millis and serialization)